src/components/restaurant/Restaurant.js

```diff
@@ -41,18 +48,15 @@ class Restaurant extends React.Component {
     }
 
     setRestaurantData = (restaurantData) => {
-        const restaurantDetails =
-            restaurantReader.restaurantDetails(restaurantData);
+        const { dispatch } = this.props;
+        const restaurantDetails = restaurantReader.restaurantDetails(restaurantData);
         const foodList = restaurantReader.foodList(restaurantData);
-        this.setState({
-            restaurantDetails,
-            foodList,
-        });
+        dispatch(setRestaurantDetails(restaurantDetails));
```

Comment:
this is fine but instead what you can do pass the second argument to the connect function an object which will contain all the action creators and you can access it using props object and you can call them directly